eng/code-quality-reports/src/test/java/com/azure/tools/checkstyle/checks/NoImplInPublicApiTest.java

```diff
@@ -0,0 +1,374 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport;
+import com.puppycrawl.tools.checkstyle.Checker;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.util.Arrays;
+
+import static com.azure.tools.checkstyle.checks.NoImplInPublicAPI.EXTENDS_TYPE_ERROR;
+import static com.azure.tools.checkstyle.checks.NoImplInPublicAPI.IMPLEMENTS_TYPE_ERROR;
+import static com.azure.tools.checkstyle.checks.NoImplInPublicAPI.RETURN_TYPE_ERROR;
+import static com.azure.tools.checkstyle.checks.NoImplInPublicAPI.TYPE_PARAM_TYPE_ERROR;
+
+public class NoImplInPublicApiTest extends AbstractModuleTestSupport {
+    private Checker checker;
+
+    @Before
+    public void prepare() throws Exception {
+        checker = createChecker(createModuleConfig(NoImplInPublicAPI.class));
+    }
+
+    @After
+    public void cleanup() {
+        checker.destroy();
+    }
+
+    @Override
+    protected String getPackageLocation() {
+        return "com/azure/tools/checkstyle/checks/NoImplInPublicApiCheck";
+    }
+
+    @Test
+    public void staticInitializerDoesNotCountAsApi() throws Exception {
+        String[] expected = new String[0];
+        verify(checker, getPath("StaticInitializer.java"), expected);
+    }
+
+    @Test
+    public void gettersNeedToCheckForImplementation() throws Exception {
+        String[] expected = {
+            expectedErrorMessage(42, 12, String.format(RETURN_TYPE_ERROR, "AnImplementationClass")),
+            expectedErrorMessage(46, 15, String.format(RETURN_TYPE_ERROR, "AnImplementationClass")),
+            expectedErrorMessage(59, 36, String.format(RETURN_TYPE_ERROR,
+                "com.azure.implementation.AnImplementationClass")),
+            expectedErrorMessage(63, 39, String.format(RETURN_TYPE_ERROR,
+                "com.azure.implementation.AnImplementationClass"))
+        };
+
+        verify(checker, getPath("Getters.java"), expected);
+    }
+
+    @Test
+    public void publicClassImplementsPublicApi() throws Exception {
+        File file = TestUtils.createCheckFile("publicClassImplementsPublicApi", Arrays.asList(
+            "package com.azure;",
```

Comment:
We should have a test for the implementation package to validate that it's a no-op.